### PR TITLE
ASC file support

### PR DESCRIPTION
**Pull request by bcumming**
Add support for Neurolucida ASCII files

* update arbor submodule to the latest version that has support for .asc files
* integrate `arborio::load_asc` into the gui
* a few small fixes to match the new Arbor API
* reduce the number of includes in headers.

### CODE REVIEW

File: src/loader.cpp [hunk 1]
```diff
@@ -36,7 +46,10 @@ loaded_morphology load_neuroml(const std::filesystem::path &fn) {
 }
 
 loaded_morphology load_asc(const std::filesystem::path &fn) {
-    loaded_morphology result;
+    auto m = arborio::load_asc(fn);
+    loaded_morphology result{.morph=m.morphology};
+    for (const auto &[k, v]: m.labels.regions()) result.regions.emplace_back(k, to_string(v));
```
Comment (thorstenhater) [inline src/loader.cpp:51]: A tiny formatting nitpick: `auto& []` instead of `auto &[]`.

**Review by thorstenhater — CHANGES_REQUESTED** [review]
Looks good, a tiny formatting suggestion.

File: src/loader.cpp (hunk 1 above)
Comment (bcumming) [reply src/loader.cpp:51]: Ha, serves me right for cut and pasting from your code above ;)

**Review by thorstenhater — APPROVED** [review]
:+1:

File: src/loader.cpp (hunk 1 above)
Comment (thorstenhater) [reply src/loader.cpp]: So true.

**Pull request merged**